data-raw/admiral_verify_templates.R

```diff
@@ -0,0 +1,243 @@
+  # gather all templates for this pkg (12 found) ----
+  templates <- list.files(path$template_dir, pattern = "ad_")
+
+  # from templates  generate vector of adam_names
+  adam_names <- vapply(templates, function(x) gsub("ad_|\\.R", "", x),
+    USE.NAMES = FALSE, character(length = 1)
+  )
+
+  # check
+  if (length(templates) != length(adam_names)) stop("Number of templates and adam_names differ")
+
+  # templates is a named chr[]
+  names(templates) <- adam_names
+
+  # per Ben, ignore "adlbhy"
+  adam_names <- adam_names[adam_names != "adlbhy"]
+
+  # download, save prior ADaMs from pharmaverseadam
+
+  download_adam_old(adam_names, path = path$adam_old_dir)
+
+  # keys for diffdf
+  keys <- teal.data::default_cdisc_join_keys
+
+  # keys is named list of keys, each element (for each adam_name) and is chr[] of keys
+  keys <- sapply(toupper(adam_names), function(e) unname(keys[[e]][[e]]), USE.NAMES = TRUE)
+  names(keys) <- tolower(names(keys))
+  keys
+
+
+  # finally, construct a named list object: each element (adam_names) holds a template and keys
+  obj <- lapply(adam_names, function(e) {
+    list(
+      "template" = templates[[e]],
+      "keys" = keys[[e]]
+    )
+  })
+  names(obj) <- adam_names
+
+  # done !
+  obj
+  names(obj)
+
+  # for TESTING, otherwise will be ALL adam_names
+  adam_names <- c("adsl", "adae")
+
+  sprintf("---- Run templates\n")
+  compare_list <- purrr::map(adam_names, .progress = TRUE, function(adam) {
+    cat("---- Template running for ", adam, "\n")
+    run_template(adam, dir = path$template_dir)
+    dataset_new <- get_dataset_new(adam, path$cache_dir)
+    dataset_old <- get_dataset_old(adam, path$adam_old_dir)
+    compare(
+      base = dataset_old,
+      compare = dataset_new,
+      keys = obj[[adam]]$keys,
+      file = paste0(path$diff, "/", adam, ".txt")
+    )
+  })
+
+  # TODO:  cleanup?
+
+  # run AFTER verify_templates() completes, otherwise will not print
+  display_diff(dir = path$diff)
+}
+
+#------------------------  helper functions
+
+#' Display Results of running diffdf
+#'
+#' Reads all text files in a specified directory and returns their contents in a named list.
+#' Each entry in the list contains the lines of one file as a character vector, and the
+#' entry is named after the file.
+#'
+#' @param dir A character string specifying the path to the directory containing the text files.
+#'
+#' @return A named list. Each element is a character vector giving the lines of a file;
+#'   the names of the list correspond to the individual filenames.
+#'
+#' @examples
+#' \dontrun{
+#' # Read contents of all files in "logs" directory
+#' file_contents <- display_diff("logs")
+#' print(names(file_contents)) # Shows the filenames
+#' print(file_contents[[1]]) # Shows contents of the first file
+#' }
+#'
+#' @export
+display_diff <- function(dir = NULL) {
+  files <- list.files(dir, full.names = TRUE)
+  contents <- lapply(files, readLines)
+  names(contents) <- basename(files)
+  contents
```

Comment:
Could we make the output a bit nicer? For example by using:
```suggestion
  map2(
    names(contents), contents,
    function(name, content) {
      cli::cli_h1(paste("Differences found for", str_replace_all(name, ".txt", ""), "\n"))
      cat(paste(content, collapse = "\n"))
    }
  )
  invisible(NULL)
```